internal/provisioner/plain/controllers/bundle_controller.go

```diff
@@ -392,3 +393,67 @@ func (r *BundleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.ConfigMap{}).
 		Complete(r)
 }
+
+func (r *BundleReconciler) bundleImagePod(pod *corev1.Pod, source rukpakv1alpha1.ImageSource) *corev1.Pod {
+	if len(pod.Spec.InitContainers) != 1 {
+		pod.Spec.InitContainers = make([]corev1.Container, 1)
+	}
+
+	pod.Spec.InitContainers[0].Name = "install-unpacker"
+	pod.Spec.InitContainers[0].Image = r.UnpackImage
+	pod.Spec.InitContainers[0].ImagePullPolicy = corev1.PullIfNotPresent
+	pod.Spec.InitContainers[0].Command = []string{"cp", "-Rv", "/unpack", "/util/unpack"}
+	pod.Spec.InitContainers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/util"}}
+
+	if len(pod.Spec.Containers) != 1 {
+		pod.Spec.Containers = make([]corev1.Container, 1)
+	}
+
+	pod.Spec.Containers[0].Name = bundleUnpackContainerName
+	pod.Spec.Containers[0].Image = source.Ref
+	pod.Spec.Containers[0].Command = []string{"/bin/unpack", "--bundle-dir", "/"}
+	pod.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/bin"}}
+
+	return pod
+}
+
+func (r *BundleReconciler) bundleGitRepoPod(pod *corev1.Pod, source rukpakv1alpha1.GitSource) (*corev1.Pod, error) {
+	// Note: use host networking (hack) so we can resolve https://github.com references in the
+	// Bundle's spec.Source.Git.Repository. Maybe we should revisit having the user inline https://github.com?
+	pod.Spec.HostNetwork = true
+
+	if len(pod.Spec.InitContainers) != 2 {
+		pod.Spec.InitContainers = make([]corev1.Container, 2)
+	}
+
+	pod.Spec.InitContainers[0].Name = "install-unpacker"
+	pod.Spec.InitContainers[0].Image = r.UnpackImage
+	pod.Spec.InitContainers[0].ImagePullPolicy = corev1.PullIfNotPresent
+	pod.Spec.InitContainers[0].Command = []string{"cp", "-Rv", "/unpack", "/util/unpack"}
+	pod.Spec.InitContainers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/util"}}
+
+	// Note: initContainer so we can ensure the repository has been cloned
+	// at the bundle.Spec.Source.Git.Ref before we unpack the Bundle contents
+	// that are stored in the repository.
+	pod.Spec.InitContainers[1].Name = "clone-repository"
+	pod.Spec.InitContainers[1].Image = r.GitClientImage
```

Comment:
nit: I think that the `GitClientImage` could end up feeling like some hidden configuration given that it is a command line flag for a binary most people won't be running directly. Maybe some documentation in the code could clear that up? We'll be updating the docs in a later PR right?

Reply:
That's true that this field is a bit hidden but it's largely an implementation detail that I'm not sure we want to leak in the user documentation. I feel like this field will be largely used when we start hitting rate-limiting issues from dockerhub and have to mirror the git image to quay. 

This git-pod based implementation is pretty heavily coupled to the plain provisioner so it's not clear this would have immediate value to other provisioners either. 

I can add a comment in the code to provide more context though 👍